projects/angular/src/utils/list-key-navigation/key-navigation-list.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { AfterViewInit, Directive, ElementRef, Inject, Input, OnDestroy, Renderer2 } from '@angular/core';
+import {
+  focusElement,
+  getFlattenedFocusableItems,
+  initializeKeyListItems,
+  KeyNavigationCode,
+  setActiveKeyListItem,
+  validKeyNavigationCode,
+} from '@cds/core/internal';
+
+import { FocusableElementsToken } from './providers/focusable-items';
+import { ClrListKeyNavigationConfigToken } from './providers/key-navigation-list.config';
+import { getNextKeyListItem } from './utils';
+
+export interface KeyNavigationListConfig {
+  keyListItems: string;
+  layout: 'both' | 'horizontal' | 'vertical';
+  manageFocus: boolean;
+  manageTabindex: boolean;
+  loop: boolean;
+  dir: string | null;
+}
+
+@Directive({
+  selector: '[clrKeyNavigationList]',
+})
+export class ClrKeyNavigationList implements AfterViewInit, OnDestroy {
+  private config: KeyNavigationListConfig;
+
+  private unlisteners: (() => void)[] = [];
+
+  @Input()
+  set configuration(config: Partial<KeyNavigationListConfig>) {
+    this.config = {
+      ...this.defaultConfig,
+      dir: this.host.nativeElement.getAttribute('rtl'),
```

Comment:
I think this should be reading the `dir` attribute. https://www.w3schools.com/tags/att_global_dir.asp

Also: What if the `dir` attribute changes at after this directive is instantiated or the `configuration` input is set? What if the `dir` attribute is set on a parent element?

I think this attribute should be read when the arrow keys are pressed in case it has changed. Also, as the `dir` attribute should be inherited, the code should look up the DOM tree until it finds it or hits the root `html` element.